### PR TITLE
Encrypt the passwords with scrypt

### PR DESCRIPTION
**Pull request by np**

### CODE REVIEW

File: src/Arbitrary.hs [hunk 1]
```diff
@@ -638,8 +639,9 @@ instance Arbitrary InitialPassword where
     shrink = gshrink
 
 instance Arbitrary EncryptedPassword where
-    arbitrary = garbitrary
-    shrink    = gshrink
+    arbitrary = mk <$> arb <*> arb
+      where
+        mk salt pass = ScryptEncryptedPassword . getEncryptedPass $ encryptPass' (Salt salt) (Pass pass)
```
Comment (fisx) [inline src/Arbitrary.hs]: i wonder why you don't want the other two cases, but i don't feel strongly either way.

File: src/Types.hs [hunk 2]
```diff
@@ -29,6 +29,7 @@ where
 import Control.Lens
 import Control.Monad
 import Control.Monad.Trans.Except (ExceptT, runExceptT)
+import Crypto.Scrypt
```
Comment (fisx) [inline src/Types.hs]: (we should really break up this module...)

**Comment by fisx** [comment]
LGTM & works.  will merge when green.  thanks!

File: src/Arbitrary.hs (hunk 1 above)
Comment (np) [inline src/Arbitrary.hs]: Which other two?

File: src/Frontend/Page/User.hs [hunk 3]
```diff
@@ -115,7 +115,8 @@ instance FormPage PageUserSettings where
             <*> ("old-password"  .:
                     -- while we need to check that the old password is the correct
                     -- one, we do not need to validate it against the rules for new passwords.
-                    DF.checkM "Das alte Passwort ist nicht korrekt" checkUserPassword
+                    -- TODO: why can't the user provide a Nothing here to circumvent the check?
```
Comment (np) [inline src/Frontend/Page/User.hs]: The user can provide nothing but `checkPwdAllOrNothing` is checking for that.

File: src/Arbitrary.hs (hunk 1 above)
Comment (fisx) [inline src/Arbitrary.hs]: my mistake.  this is the type for encrypted passwords only, and it is contained in the `UserPass`, where there are other constructors.  i got the two confused.

File: src/Frontend/Page/User.hs (hunk 3 above)
Comment (fisx) [inline src/Frontend/Page/User.hs]: ok makes sense.  i'll add a comment about that.  (unless you want to do it...?)

Comment (np) [inline src/Frontend/Page/User.hs]: To me the comment line 98 was enough.

**Pull request merged**